nisystemlink/clients/result/models/_query_results_request.py

```diff
@@ -0,0 +1,141 @@
+    this list. For example, "@0" in the filter expression will be replaced with the element at the
+    zeroth index of the substitutions list.
+    """
+
+
+class QueryProductsBase(JsonModel):
+    product_filter: Optional[str] = None
+    """
+    The product query filter in Dynamic Linq format.
+    Allowed properties in the filter are:
+    - `id`: String for the global identifier of the product
+    - `partNumber`: String representing the part number of the product
+    - `name`: String of the product name
+    - `family`: String for the product family
+    - `updatedAt`: ISO-8601 formatted UTC timestamp indicating when the product was last updated.
+    - `keywords`: A list of keyword strings
+    - `properties`: A dictionary of additional string to string properties
+    - `fileIds`: A list of string ids for files stored in the file service (`/nifile`)
+    See [Dynamic Linq](https://github.com/ni/systemlink-OpenAPI-documents/wiki/Dynamic-Linq-Query-Language)
+    documentation for more details.
+    `"@0"`, `"@1"` etc. can be used in conjunction with the `substitutions` parameter to keep this
+    query string more simple and reusable.
+    """
+
+    product_substitutions: Optional[List[str]] = None
+    """String substitutions into the `filter`.
+    Makes substitutions in the query filter expression. Substitutions for the query expression are
+    indicated by non-negative integers that are prefixed with the "at" symbol. Each substitution in
+    the given expression will be replaced by the element at the corresponding index (zero-based) in
+    this list. For example, "@0" in the filter expression will be replaced with the element at the
+    zeroth index of the substitutions list.
+    """
+
+
+class QueryResultsRequest(QueryResultsBase, QueryProductsBase):
+
+    order_by: Optional[ResultField] = Field(None, alias="orderBy")
+    """Specifies the fields to use to sort the results.
+    By default, results are sorted by `id`
+    """
+    order_by_key: Optional[str] = Field(None, alias="orderByKey")
+    """Specifies the property to use to sort the results when ordering by PROPERTIES.
+    Results that do not contain the orderByKey will be considered the smallest value.
+    """
+    order_by_comparison_type: Optional[ComparisonType] = Field(
+        None, alias="orderByComparisonType"
+    )
+    """An enumeration of comparison types that can be used for ordered queries.
+    For non-DEFAULT comparisons, values that cannot be converted will be considered the smallest value.
+    """
+    descending: Optional[bool] = None
+    """Specifies whether to return the results in descending order.
+    By default, this value is `false` and results are sorted in ascending order.
+    """
+    take: Optional[int] = None
+    """Maximum number of results to return in the current API response.
+    Uses the default if the specified value is negative. The default value is `1000` results.
+    """
+    continuation_token: Optional[str] = None
```

Comment:
We prefer to base off of `WithPaging` to get the continuation token field. That will help when we have a more reusable paging implementation.